RasaHQ/rasa_core
# Broker endpoint configuration

**Goal**:
In preparation for https://github.com/RasaHQ/rasa_platform/issues/265 it is required to outsource the creation of the broker because you would always need to write custom code otherwise.

**Proposed changes**:
- add a section `event_broker` to the endpoint configuration
- initialise the event broker from the values in the configuration
- add a directory `test_endpoints` to create directory structure which is able to hold different examples of endpoint configurations.

**To test**:
1. Start a message broker. The following command runs RabbitMQ with a management interface which you can then access on <http://localhost:15672>
```
docker run -e RABBITMQ_DEFAULT_USER=username -e RABBITMQ_DEFAULT_PASS=password -p 5672:5672 -p 15672:15672 rabbitmq:3-management
```
2. Start rasa core and you use a endpoint configuration file, e.g. the one in `/data/test_endpoints/event_broker_endpoint.yml`
```
python -m rasa_core.run -d <dir of rasa core model> --endpoints data/test_endpoints/event_broker_endpoint.yml
```
3. Create an event by sending a message to the chatbot (you can then check <http://localhost:15672> for this event)

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
